google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminSettings.java

```diff
@@ -84,19 +89,45 @@ public Builder toBuilder() {
     return new Builder(this);
   }
 
-  /** Returns a new builder for this class. */
+  /**
+   * Returns a new builder for this class.
+   *
+   * <p>If emulator configuration provided in BIGTABLE_EMULATOR_HOST environment variable then it
+   * creates a builder preconfigured to connect to Bigtable using emulator hostname and port number.
+   */
   public static Builder newBuilder() {
+    String hostAndPort = System.getenv(BIGTABLE_EMULATOR_HOST_ENV_VAR);
+    if (hostAndPort != null && !hostAndPort.isEmpty()) {
+      int port;
+      try {
+        port = Integer.parseInt(hostAndPort.substring(hostAndPort.lastIndexOf(":") + 1));
+        return newBuilderForEmulator(hostAndPort.substring(0, hostAndPort.lastIndexOf(":")), port);
+      } catch (NumberFormatException ex) {
+        throw new RuntimeException(
+            "Invalid port in "
+                + BIGTABLE_EMULATOR_HOST_ENV_VAR
+                + " environment variable: "
+                + hostAndPort);
+      }
+    }
     return new Builder();
   }
 
-  /** Create a new builder preconfigured to connect to the Bigtable emulator. */
+  /** Create a new builder preconfigured to connect to the Bigtable emulator with port number. */
   public static Builder newBuilderForEmulator(int port) {
-    Builder builder = newBuilder().setProjectId("fake-project").setInstanceId("fake-instance");
+    return newBuilderForEmulator("localhost", port);
+  }
+
+  /**
+   * Create a new builder preconfigured to connect to the Bigtable emulator with host & port number.
+   */
+  public static Builder newBuilderForEmulator(String hostname, int port) {
+    Builder builder = new Builder().setProjectId("fake-project").setInstanceId("fake-instance");
```

Comment:
I don't think we should be setting the project instance ids to stay consistent with other languages

Reply:
For Java, we have projectId & instanceId required. Shall I remove the `Preconditions` checks for projectId & InstanceId?
 I would vote to keep these checks and continue with "fake-project" & "fake-instance".

Reply:
I'm not sure what you mean by that. I'm picturing something along the lines of:

```
BigtableTableAdminSettings.newBuilderForEmulator("localhost", 1234).setProjectId("blah").setInstanceId("blah2").build()
```

I think the other client languages require something similar

Reply:
Thanks for the explanation, I mistook it as emulator without any projectId & instanceId. Have updated with the suggestion. please have a look.